### ISSUE TITLE
cannot encrypt to my OpenPGP certificate D477040C70C2156A5C298549BB7E9101495E6BF7 (rejects cert when encountering unknown sigtype)

### ISSUE THREAD

**Issue by dkg** [vanitasvitae/pgpainless#458]
My OpenPGP certificate as distributed by keys.openpgp.org contains a [1pa3pc](https://datatracker.ietf.org/doc/draft-dkg-openpgp-1pa3pc/) indicator of approved third-party certifications.  While this is not an IETF-ratified specification, other sop implementations are capable of skipping over this unknown signature type and processing the rest of the certificate anyway.

```
0 dkg@bob:/tmp/cdtemp.e5AGdx$ wget -q https://keys.openpgp.org/vks/v1/by-fingerprint/D477040C70C2156A5C298549BB7E9101495E6BF7
0 dkg@bob:/tmp/cdtemp.e5AGdx$ pgpainless-cli version --extended
PGPainless-SOP 1.6.8
https://codeberg.org/PGPainless/pgpainless/src/branch/master/pgpainless-sop

Implementation of the Stateless OpenPGP Protocol Version 07
https://datatracker.ietf.org/doc/html/draft-dkg-openpgp-stateless-cli-07

Based on pgpainless-core 1.6.8
https://pgpainless.org

Using Bouncy Castle 1.77
https://www.bouncycastle.org/java.html
0 dkg@bob:/tmp/cdtemp.e5AGdx$ echo test | pgpainless-cli encrypt D477040C70C2156A5C298549BB7E9101495E6BF7 
Signature type 0x16 appears to be invalid.
1 dkg@bob:/tmp/cdtemp.e5AGdx$ echo test | sqop encrypt D477040C70C2156A5C298549BB7E9101495E6BF7 
-----BEGIN PGP MESSAGE-----

wV4DOAJNcYq6PzsSAQdAlPKqDfR2vuJVGI8G7TkG1mq7xYQHmuTqO2f2kt4jv2cw
RS4LBs/l0S0Wh9sOgJqd5Mcf/1PU863T0+rNcvQrnCnB7OMQ0neRAzEmESIstSy9
0jYBj5e/wfY9hpgfZEZCPhxsMVchsxmUcMKpX1eZTY7c0QQu68iPI9ibBt4WEw00
1wX5QovXzYc=
=50Ts
-----END PGP MESSAGE-----
0 dkg@bob:/tmp/cdtemp.e5AGdx$ echo test | rsop encrypt D477040C70C2156A5C298549BB7E9101495E6BF7 
-----BEGIN PGP MESSAGE-----

wV4DOAJNcYq6PzsSAQdAh/d6BEzyPqBdPLfV3Hu62oIENfkraBLXLJmt/xt42SQw
hTLaOlI3GB98ozTpbmYhcOjh2GqP6CdRvsczpexFUggucR1qZgYQiAEAv4P/dvyn
0jYBbwI46zyjO1JMlBibGhfsJmvRCd4D+7giOsNh4LjnOQEYmWIIa6T3rxT/GHH+
32ZFzRkmJAY=
=EEBR
-----END PGP MESSAGE-----
0 dkg@bob:/tmp/cdtemp.e5AGdx$ echo test | gosop encrypt D477040C70C2156A5C298549BB7E9101495E6BF7 
-----BEGIN PGP MESSAGE-----

wV4DOAJNcYq6PzsSAQdA0bCUEJKESiYNNN6eE+D0+UH1AaOc5CQAzYe1t4W4qXIw
PIePxW6Je6G+XylVigKY8AdhHwq7lj0T7tdg2pD5Fz0qlIJnrTTqRqQCHZa1OWLB
0jYBvYeEGjVDlDN1txZeEH3POixLLGkCUTvjyeC2olZXDW59T3VjEMWDw7Q+rKEb
Gc36jkw91lo=
=Faxh
-----END PGP MESSAGE-----
0 dkg@bob:/tmp/cdtemp.e5AGdx$ 
```

**Comment by vanitasvitae** [comment]
I think it should be easy to fix this in PGPainless.
Thanks for the report!

**Comment by vanitasvitae** [comment]
```
-----BEGIN PGP MESSAGE-----

wV4DOAJNcYq6PzsSAQdAVQj0KqOUQQG6zoF1vIQFGQj3hIFwiAVnG0nDnufM3Qgw
N3p4LgaxSideTzVEN/gq6K+p8njURMLO0V5odDuKyYR5pYKGNAXKx6THiKrm+GlQ
0jcBpzMy8bTIsFOOkVmsmggg6T2sh3/KQj2Fjvu2I0D+UabHymr9HUQkmX6VQqXi
HcsddNu6Ts4t
=0++c
-----END PGP MESSAGE-----
```
;)

Edit: I fixed this in `main` and will see if I can backport the fix to the `1.6` branch.
